whatlies/embeddingset.py

```diff
@@ -580,6 +592,94 @@ def plot_correlation(self, metric=None):
         # Rotate the tick labels and set their alignment.
         plt.setp(ax.get_xticklabels(), rotation=90, ha="right", rotation_mode="anchor")
 
+    def plot_similarity(self, metric="cosine", norm=False):
+        """
+        Make a similarity plot. Shows you the similarity between all the word embeddings in the set.
+
+        Arguments:
+            metric: `cosine` or `correlation`
+            norm: normalise the embeddings before calculating the distance
+
+        Usage:
+
+        ```python
+        from whatlies.language import SpacyLanguage
+        lang = SpacyLanguage("en_core_web_sm")
+
+        names = ['red', 'blue', 'green', 'yellow', 'cat', 'dog', 'mouse', 'rat', 'bike', 'car']
+        emb = lang[names]
+        emb.plot_similarity()
+        emb.plot_similarity(metric='correlation')
+        ```
+        """
+        allowed_metrics = ["cosine", "correlation"]
+        if metric not in allowed_metrics:
+            raise ValueError(
+                f"The `metric` argument must be in {allowed_metrics}, got: {metric}."
+            )
+
+        vmin, vmax = 0, 1
+        X = self.to_X(norm=norm)
+        if metric == "cosine":
+            similarity = cosine_similarity(X)
+        if metric == "correlation":
+            similarity = np.corrcoef(self.to_X())
+
+        fig, ax = plt.subplots()
+        plt.imshow(similarity, cmap=plt.cm.get_cmap(), vmin=-vmin, vmax=vmax)
+        plt.xticks(range(len(self)), self.embeddings.keys())
+        plt.yticks(range(len(self)), self.embeddings.keys())
+        plt.colorbar()
+
+        # Rotate the tick labels and set their alignment.
+        plt.setp(ax.get_xticklabels(), rotation=90, ha="right", rotation_mode="anchor")
+
+    def plot_distance(self, metric="cosine", norm=False):
+        """
+        Make a similarity plot. Shows you the similarity between all the word embeddings in the set.
+
+        Arguments:
+            metric: `cosine`, `correlation` or `euclidean`
+            norm: normalise the vectors before calculating the distances
+
+        Usage:
+
+        ```python
+        from whatlies.language import SpacyLanguage
+        lang = SpacyLanguage("en_core_web_sm")
+
+        names = ['red', 'blue', 'green', 'yellow', 'cat', 'dog', 'mouse', 'rat', 'bike', 'car']
+        emb = lang[names]
+        emb.plot_distance(metric='cosine')
+        emb.plot_distance(metric='euclidean')
+        emb.plot_distance(metric='correlation')
+        ```
+        """
+        allowed_metrics = ["cosine", "correlation", "euclidean"]
+        if metric not in allowed_metrics:
+            raise ValueError(
+                f"The `metric` argument must be in {allowed_metrics}, got: {metric}."
+            )
+
+        vmin, vmax = 0, 1
+        X = self.to_X(norm=norm)
+        if metric == "cosine":
+            distances = cosine_distances(X)
+        if metric == "correlation":
+            distances = 1 - np.corrcoef(self.to_X())
```

Comment:
The values of Pearson corr coefficients are between -1 and 1. So do you want to scale the values to [0, 1] considering that you have set `vmin, vmax = 0, 1`?

Reply:
Well spotted!